### ISSUE TITLE
Decision Trees

### ISSUE THREAD

**Issue by oke-aditya** [madewithml/hacktoberfest#10]
## 🚀 Tutorial / New Example Request

Decision Trees are quite fundamental (I think they are not covered in Andrew Ng's course though)

Can we have a complete in-depth tutorial

**Describe the solution you'd like**
1. Keep the guide simple. Do not involve a lot of maths. Explain the terms `gini` index `entropy` in simple form.
2. Choose a simple dataset. Like iris from sklearn. And show how easy it is to train a decision tree.
3. Walk through how it works E.g. CD Tree, CART, etc algorithms. A brief paragraph about each is sufficient.
4. There are a lot of parameters in sklearn, explain them and show how they affect while trianing. E.g. `ccp_alpha` is cost computation pruning to reduce the overfit issue.
5. How decision trees are extended to regression.
6. Vizualize the tree using Graphviz, or other library and show how it works, how the split occurs.

More ideas welcome 👍 

**Describe alternatives you've considered**
I think `tf.estimtator` too has decision tree but not sure, people can have a look.

**Comment by SauravMaheshkar** [comment]
Please assign this to me. Shall I mention the theory in the README.md or should I make a separate section for it in a Jupyter Notebook?

**Comment by oke-aditya** [comment]
You can mention it within Jupyter Notebook itself. Divide the Jupyter notebook into proper sections.

One of my proposal to explain ML models is as follows.

1. Data

2 Task

3 Model

4 Loss function

5 Learning algorithm

6 Evaluation Metrics

This leads to clean presentation of ML models.

Have a look here :-)

https://medium.com/machine-learning-magazine/a-guide-to-supervised-learning-b9c227a71155

Feel free to use your own style of writing though. It isn't something to enforce, but ensure to maintain a good flow.

**Comment by SauravMaheshkar** [comment]
Hey, 

I've been working on this [**Notebook**](https://colab.research.google.com/drive/1tImSKOiA2jsZfUBexDjPY2lEXj8SnaHx?usp=sharing).  Please tell suggestions and improvements.

**Comment by oke-aditya** [comment]
Hey @SauravMaheshkar can you raise a PR, we have ReviewNB app that makes reviewing Jupyter Notebooks very simple.

I can comment on it there as well.
